host/xtest/Makefile

```diff
@@ -4,6 +4,17 @@
 ifeq ($O,)
 $(error output path should be specified when calling this makefile)
 endif
+# Remove trailing / and /. in output path, rmdir does not like them
+out-dir := $(shell s=$O; while [ true ]; do \
+            [ "$$s" == "/" ] && break; \
+            s=`echo "$$s" | sed s\/\\\/$$\/\/ --`; \
+            [ "$$s" == "/" ] && break; \
+            s=`echo "$$s" | sed s\/\\\/\.$$\/\/ --`; \
+            echo $$s | grep --quiet /$$ && continue; \
+            echo $$s | grep --quiet /\.$$ && continue; \
+	    break; \
+        done; echo $$s)
+
```

Comment:
you mean adding a output_dir.mk in optee_xtest and including from each concerned Makefile in optee_test ? why not.